NoSuchFileException for function.zip in amazon-lambda

**Describe the bug**
When using the quarkus-amazon-lambda dependency for a native lambda function, when doing a `mvn package -Pnative` for the first time, the process fails with a NoSuchFileException on the function.zip. The weird thing is that it is trying to delete the file, while it should be creating it. 

**Expected behavior**
The process executes without errors and has created all the necessary files, function.zip inclusive. 

**Actual behavior**
The process fails on a NoSuchFileException. Almost all files seem to be created (native executable, manage.sh, sam-files, bootstrap-example, ... but no function.zip


**To Reproduce**
Steps to reproduce the behavior:
1. Create a simple lambda using the quarkus-amazon-lambda plugin 
2. Make sure the necessary profile for native images is available 
3. Execute a mvn package -Pnative 

**Configuration**
```properties
Not applicable 

```

**Screenshots**
Not available

**Fix** 
A fork with a new branch that contains a bug-fix has already been made and a pull request has been opened: https://github.com/quarkusio/quarkus/pull/8884

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin Lestat.local 19.4.0 Darwin Kernel Version 19.4.0: Wed Mar  4 22:28:40 PST 2020; root:xnu-6153.101.6~15/RELEASE_X86_64 x86_64
 - Output of `java -version`: java version "12.0.2" 2019-07-16
Java(TM) SE Runtime Environment (build 12.0.2+10)
Java HotSpot(TM) 64-Bit Server VM (build 12.0.2+10, mixed mode, sharing)
 - GraalVM version (if different from Java): not applicable: native build using docker
 - Quarkus version or git rev: 1.4.0.Final, but it also seems to be on the master branch's head: a8f3fccac1cbf0f34d5f3262d6a472e7c42ed509
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

**Additional context**
Stacktrace: 

> [ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.4.0.Final:build (default) on project privacy-gate-cookie: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.amazon.lambda.deployment.FunctionZipProcessor#nativeZip threw an exception: java.lang.IllegalStateException: java.nio.file.NoSuchFileException: /Users/stef/projects/lambda-cookie/target/function.zip
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:940)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR]         at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2019)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1558)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1449)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:830)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:479)
[ERROR] Caused by: java.nio.file.NoSuchFileException: /Users/stef/projects/lambda-cookie/target/function.zip
[ERROR]         at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
[ERROR]         at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
[ERROR]         at java.base/sun.nio.fs.UnixFileSystemProvider.implDelete(UnixFileSystemProvider.java:249)
[ERROR]         at java.base/sun.nio.fs.AbstractFileSystemProvider.delete(AbstractFileSystemProvider.java:105)
[ERROR]         at java.base/java.nio.file.Files.delete(Files.java:1145)
[ERROR]         at io.quarkus.amazon.lambda.deployment.FunctionZipProcessor.nativeZip(FunctionZipProcessor.java:105)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:567)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:931)
[ERROR]         ... 7 more
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

--- a/extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java
+++ b/extensions/amazon-lambda/common-deployment/src/main/java/io/quarkus/amazon/lambda/deployment/FunctionZipProcessor.java
@@ -102,7 +102,7 @@ public class FunctionZipProcessor {
         Path zipDir = findNativeZipDir(target.getOutputDirectory());
 
         Path zipPath = target.getOutputDirectory().resolve("function.zip");
-        Files.delete(zipPath);
+        Files.deleteIfExists(zipPath);
         try (ZipArchiveOutputStream zip = new ZipArchiveOutputStream(zipPath.toFile())) {
             String executableName = "bootstrap";
             if (zipDir != null) {

